ts/features/bonus/cgn/store/reducers/bucket.ts

```diff
@@ -0,0 +1,50 @@
+import { getType } from "typesafe-actions";
+import { Action } from "../../../../../store/actions/types";
+import { GlobalState } from "../../../../../store/reducers/types";
+import { NetworkError } from "../../../../../utils/errors";
+import {
+  remoteError,
+  remoteLoading,
+  remoteReady,
+  remoteUndefined,
+  RemoteValue
+} from "../../../bpd/model/RemoteValue";
+import { cgnCodeFromBucket, DiscountBucketCode } from "../actions/bucket";
+
+export type CgnBucketState = {
+  data: RemoteValue<DiscountBucketCode, NetworkError>;
+};
+
+const INITIAL_STATE: CgnBucketState = {
+  data: remoteUndefined
+};
+
+const reducer = (
+  state: CgnBucketState = INITIAL_STATE,
+  action: Action
+): CgnBucketState => {
+  switch (action.type) {
+    case getType(cgnCodeFromBucket.request):
+      return {
+        ...state,
+        data: remoteLoading
+      };
+    case getType(cgnCodeFromBucket.success):
+      return {
+        ...state,
+        data: remoteReady(action.payload)
+      };
+    case getType(cgnCodeFromBucket.failure):
+      return {
+        ...state,
+        data: remoteError(action.payload)
+      };
+  }
+  return state;
+};
+
+export default reducer;
+
+export const cgnOtpDataSelector = (
```

Comment:
```suggestion
export const cgnBucketSelector = (
```
The same as previous comment.